kernel-spark/src/main/java/io/delta/kernel/spark/snapshot/unitycatalog/SparkUnityCatalogUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.spark.snapshot.unitycatalog;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.spark.utils.CatalogTableUtils;
+import io.delta.storage.commit.uccommitcoordinator.UCCommitCoordinatorClient;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.catalog.CatalogTable;
+import org.apache.spark.sql.delta.coordinatedcommits.UCCommitCoordinatorBuilder$;
+
+/**
+ * Utility class for extracting Unity Catalog connection information from Spark catalog metadata.
+ *
+ * <p>This class isolates Spark dependencies, allowing {@link UnityCatalogAdapter} to be created
+ * without Spark if connection info is provided directly via {@link UnityCatalogConnectionInfo}.
+ */
+public final class SparkUnityCatalogUtils {
+
+  // Utility class - no instances
+  private SparkUnityCatalogUtils() {}
+
+  /**
+   * Extracts Unity Catalog connection information from Spark catalog table metadata.
+   *
+   * @param catalogTable Spark catalog table metadata
+   * @param spark SparkSession for resolving Unity Catalog configurations
+   * @return connection info if table is UC-managed, empty otherwise
+   * @throws IllegalArgumentException if table is UC-managed but configuration is invalid
+   */
+  public static Optional<UnityCatalogConnectionInfo> extractConnectionInfo(
+      CatalogTable catalogTable, SparkSession spark) {
+    requireNonNull(catalogTable, "catalogTable is null");
+    requireNonNull(spark, "spark is null");
+
+    if (!CatalogTableUtils.isUnityCatalogManagedTable(catalogTable)) {
+      return Optional.empty();
+    }
+
+    String tableId = extractUCTableId(catalogTable);
+    String tablePath = extractTablePath(catalogTable);
+
+    // Get catalog name
+    scala.Option<String> catalogOption = catalogTable.identifier().catalog();
+    String catalogName =
+        catalogOption.isDefined()
```

Comment:
Why do we need to look at spark.sessionState().catalogManager().currentCatalog().name(), it will returning `spark_catalog` as default and user specified one with "USE CATALOG" but not necessary the right one to look at